### ISSUE TITLE
CatBoost data-validation sanity check on duplicated dataset yields AUC > 0.5 and inconsistent train Logloss

### ISSUE THREAD

**Issue by guillaume-vignal** [MAIF/eurybia#82]
### Describe the bug

In a data-validation sanity check, I build a binary target by duplicating the exact same feature dataset:
- copy A: target = 0
- copy B: target = 1

Since both classes have identical feature distributions, I expect a classifier to have no discriminative power on holdout data (`AUC ~= 0.5`).

However, with `CatBoostClassifier` in the Eurybia tutorial workflow:
1. training logs show Logloss decreasing,
2. `eval_metrics(..., 'Logloss')` on train does not match what is shown during training,
3. resulting AUC is not close to 0.5.

This makes the sanity check hard to interpret.

<img width="786" height="571" alt="Image" src="https://github.com/user-attachments/assets/45710bc7-f19d-4f31-b947-b9a5aeeed5e7" />

<img width="569" height="407" alt="Image" src="https://github.com/user-attachments/assets/ce253ff5-e239-4d37-ad0b-2c56ad163cfe" />

<img width="380" height="246" alt="Image" src="https://github.com/user-attachments/assets/5966d3fe-ff88-480c-8f1a-bac1d8e8c136" />

### Minimal reproducible example

```python
import pandas as pd
import catboost
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_auc_score
from shapash.data.data_loader import data_loading
from category_encoders import OrdinalEncoder

# 1) Load data
house_df, house_dict = data_loading("house_prices")
X_df = house_df[house_df.columns.difference(["SalePrice"])].copy()

# 2) Encode categoricals
categorical_features = [col for col in X_df.columns if X_df[col].dtype == "object"]
encoder = OrdinalEncoder(
    cols=categorical_features,
    handle_unknown="ignore",
    return_df=True
).fit(X_df)
X_df = encoder.transform(X_df)

# 3) Build two identical datasets with opposite labels
X_df_1 = X_df.copy()
X_df_1["cible"] = 1
X_df["cible"] = 0
df = pd.concat([X_df, X_df_1], axis=0)

# 4) Split
train, test = train_test_split(df, test_size=0.3, random_state=42)

# 5) Train CatBoost
var = [col for col in X_df.columns if col != "cible"]
indice_cat = [i for i in range(len(var)) if var[i] in categorical_features]

train_pool = catboost.Pool(data=train[var], label=train.cible, cat_features=indice_cat)
test_pool = catboost.Pool(data=test[var], label=test.cible, cat_features=indice_cat)

model = catboost.CatBoostClassifier(
    cat_features=indice_cat,
    task_type="CPU",
    n_estimators=250,
    learning_rate=0.01,
    max_depth=3,
    l2_leaf_reg=19,
    use_best_model=True,
    loss_function="Logloss",
    eval_metric="Logloss",
    verbose=10
)

model.fit(train_pool, plot=False, eval_set=test_pool)

# 6) Compare logged vs evaluated metrics
train_logloss = model.eval_metrics(train_pool, "Logloss")
test_logloss = model.eval_metrics(test_pool, "Logloss")
print("best train iteration from eval_metrics:", int(np.argmin(train_logloss["Logloss"])))
print("best test iteration from eval_metrics:", int(np.argmin(test_logloss["Logloss"])))

# 7) Sanity AUC
y_scores_test = model.predict_proba(test[var])[:, 1]
print("test AUC:", roc_auc_score(test.cible, y_scores_test))
y_scores = model.predict_proba(train)[:,1]
roc_auc_score(train.cible, y_scores)
